internal/common/objects/MegaBike.go

```diff
@@ -12,12 +12,15 @@ type IMegaBike interface {
 	RemoveAgent(bikerId uuid.UUID)
 	GetAgents() []IBaseBiker
 	UpdateMass()
+	KickOutAgent(bikerId uuid.UUID)
 }
 
 // MegaBike will have the following forces
 type MegaBike struct {
 	*PhysicsObject
 	agents []IBaseBiker
+	kickedOutCount    int
+	currentLeaderUUID uuid.UUID
```

Comment:
this (and all the related functions) have been added already in the iteration2 branch so they shouldn't be needed 
what we were talking about for this PR was mainly for some sort of counter to keep track of how long an agent had been a leader for etc, but I think we have agreed that should be done in the server for the experiments and/or by each individual agents if they want to keep track of it